packages/theme-language-server-common/src/hover/providers/LiquidObjectAttributeHoverProvider.ts

```diff
@@ -29,14 +30,28 @@ export class LiquidObjectAttributeHoverProvider implements BaseHoverProvider {
       lookups: parentNode.lookups.slice(0, lookupIndex),
     };
 
-    const parentType = await this.typeSystem.inferType(node, ancestors[0], params.textDocument.uri);
-    if (isArrayType(parentType)) {
-      return null;
+    const objectMap = await this.typeSystem.objectMap(uri, ancestors[0]);
+    const parentType = await this.typeSystem.inferType(node, ancestors[0], uri);
+
+    if (isArrayType(parentType) || parentType === 'string') {
+      const nodeType = await this.typeSystem.inferType(parentNode, ancestors[0], uri);
+
+      // Shouldn't happen but just in case
+      if (isArrayType(nodeType) || nodeType === Unknown) return null;
+
+      // We want want `## first: `nodeType` with the docs of the nodeType
+      const entry = { ...(objectMap[nodeType] ?? {}), name: currentNode.value };
+
+      return {
+        contents: {
+          kind: 'markdown',
+          value: render(entry, nodeType),
+        },
+      };
     }
 
-    const objectMap = await this.typeSystem.objectMap(params.textDocument.uri, ancestors[0]);
     const parentEntry = objectMap[parentType];
-    if (!parentEntry) {
+    if (!parentEntry && parentType !== Unknown && parentType !== 'untyped') {
```

Comment:
am not sure about this condition anymore, should revisit. 

If I have something that is `any` and I hover over one of its properties, then its property should also be `any` and I should show a hover window.